### PR TITLE
Fix: Support HTTP requests for `FilteredPools`

### PR DESCRIPTION
**Pull request by mattverse**
cref: https://github.com/googleapis/googleapis/blob/master/google/api/http.proto

Current `FilteredPools` does not support http requests, as grpc gateway cannot do `GET` requests for non primitive types that are repeated. 

This PR brings the solution using coins string, and then parsing the coins string manually with in the grp_query.go logic. 

## Brief Changelog
- Change `FilteredPools` `MinLiquidity` field from sdk.Coins struct to string

## Testing and Verifying
Passes existing test on the grpc method, also added new test cases for the helper method.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/gamm/client/cli/query.go [hunk 1]
```diff
@@ -286,21 +283,12 @@ $ %s query gamm pools-with-filter <min_liquidity> <pool_type>
...
-				min_liquidity = coins
+				min_liquidity = args[0]
 			} else {
-				coins, err := sdk.ParseCoinsNormalized(args[0])
-				if err != nil {
-					return status.Errorf(codes.InvalidArgument, "invalid token: %s", err.Error())
-				}
-
-				min_liquidity = coins
+				min_liquidity = args[0]
```
Comment (nicolaslara) [inline x/gamm/client/cli/query.go:291]: does it make more sense to do:
```
			min_liquidity := args[0]
			var pool_type string
			if len(args) > 1 {
				pool_type = args[1]
			}
```

**Review by nicolaslara — APPROVED** [review]

**Review by p0mvn — APPROVED** [review]

**Comment by jonator** [comment]
Do you have an example of how the query should be serialized?

**Comment by mattverse** [comment]
@jonator are you asking for what an example request would look like? If so, it would look something like this
```
{
"min_liquidity": "100uosmo, 300uatom",
"pool_type": "Balancer"
}
```

**Review by ValarDragon — APPROVED** [review]
LGTM but needs API break changelog update

**Comment by ValarDragon** [comment]
Changed labels, this is query breaking, but state compatible, as its not a query exposed to state machine. (If it was we couldn't backport)

**Comment by jonator** [comment]
> @jonator are you asking for what an example request would look like? If so, it would look something like this
> 
> ```
> {
> "min_liquidity": "100uosmo, 300uatom",
> "pool_type": "Balancer"
> }
> ```

No I meant something like:
https://lcd-osmosis.keplr.app/osmosis/gamm/v1beta1/filtered_pools?pagination.limit=100&pooltype=stableswap

We can't send a body in a GET request

**Comment by mattverse** [comment]
@jonator Ah I see, it could be done using something like https://lcd-osmosis.keplr.app/osmosis/gamm/v1beta1/filtered_pools?pagination.limit=100&min_liquidity=100uosmo&pool_type=Balancer

**Pull request merged**